exporter/azureblobexporter/config.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package azureblobexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azureblobexporter"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type TelemetryConfig struct {
+	Logs    string `mapstructure:"logs"`
+	Metrics string `mapstructure:"metrics"`
+	Traces  string `mapstructure:"traces"`
+}
+
+type (
+	Container TelemetryConfig
+	BlobName  TelemetryConfig
+)
+
+type BlobNameFormat struct {
+	MetricsFormat  string            `mapstructure:"metrics_format"`
+	LogsFormat     string            `mapstructure:"logs_format"`
+	TracesFormat   string            `mapstructure:"traces_format"`
+	SerialNumRange int64             `mapstructure:"serial_num_range"`
+	Params         map[string]string `mapstructure:"params"`
+}
+
+type Authentication struct {
+	// Type is the authentication type. supported values are connection_string, service_principal, system_managed_identity and user_managed_identity
+	Type AuthType `mapstructure:"type"`
+
+	// TenantID is the tenand id for the AAD App. It's only needed when type is service principal.
+	TenantID string `mapstructure:"tenant_id"`
+
+	// ClientID is the AAD Application client id. It's needed when type is service principal or user managed identity
+	ClientID string `mapstructure:"client_id"`
+	// ClientSecret only needed when auth type is service_principal
+
+	ClientSecret string `mapstructure:"client_secret"`
+
+	// ConnectionString to the endpoint.
+	ConnectionString string `mapstructure:"connection_string"`
+}
+
+type AuthType string
+
+const (
+	ConnectionString      AuthType = "connection_string"
+	SystemManagedIdentity AuthType = "system_managed_identity"
+	UserManagedIdentity   AuthType = "user_managed_identity"
+	ServicePrincipal      AuthType = "service_principal"
+)
+
+// Config contains the main configuration options for the azure storage blob exporter
+type Config struct {
+	URL            string          `mapstructure:"url"`
```

Comment:
Please use https://pkg.go.dev/go.opentelemetry.io/collector/config/confighttp#ClientConfig instead of strict field for URL.

Reply:
Thanks @MovieStoreGuy for the careful review. I noticed there's also `auth` settings in `ClientConfig`. If I squash it, it may conflict with the `auth` setting in current config. Currently I didn't find any auth extension for Azure and I'd like to investigate/explorer about `azureauth` extension for future. And for now, is it possible to use `endpoint` string instead of `ClientConfig`, or any suggestions? Thanks!

Reply:
Right, that makes complete sense.

So at this current point, what I suggest is the following:
- Keep it as `url` for the time being as the key for the configmap
- Create an azure auth provider
- Once provider is ready, embed struct and warn users on field deprecation

There is a way that you could have the expected fields from the incoming struct and once ready, you could cleanly swap it out with minimal conversions.

I vote for the first since it is the most simple and it is fairly simple to fix up in future.

Perhaps make a note somewhere to users that this field is expected to change at some point to match the configuration standard?

Reply:
Renamed to `url` and added some descriptions. Will do below changes in future PRs.
* Create an azure auth provider
* Once provider is ready, embed struct and warn users on field deprecation